docs/introduction.rst

```diff
@@ -0,0 +1,111 @@
+Introduction
+============
+
+The XAITK-Saliency toolkit implements a class of XAI algorithms known
```

Comment:
`toolkit` -> `package`? This might help prevent overloading the term "toolkit", and I think we might also refer to `xaitk-saliency` as a package elsewhere (not 100% sure though).

Reply:
sounds good